docs/docs/manual/facets.md

```diff
@@ -0,0 +1,284 @@
+
+![A view of the expressions window with an error converting a string to a number.](/img/error.png)
+
+To output errors, ensure that you have “store error” selected for the “On error” option in the expressions window. 
+
+### Facet by null, empty, or blank
+
+Any column can be faceted for [null and/or empty cells](#cell-data-types). These can help you find cells where you want to manually enter content. “Blank” means both null values and empty values. All three facets will generate “true” and “false” facets, “true” being blank. 
+
+An empty cell is a cell that is set to contain a string, but doesn’t have any characters in it (a zero-length string). This can be a leftover from an operation that removed characters, or from manually editing a cell and deleting its contents.
+
+
+### Facet by star or flag
+
+Stars and flags offer you the opportunity to mark specific rows for yourself for later focus. Stars and flags persist through closing and opening your project, and thus can provide a different function than using a permalink to persist your facets. Stars and flags can be used in any way you want, although they are designed to help you flag errors and star rows of particular importance. 
+
+You can manually star or flag rows simply by clicking on the icons to the left of each row. You can also apply stars or flags to all matching rows by using the “All” dropdown menu and selecting “Edit rows” > “Star rows” or “Flag rows.” These operations will modify all matching rows in your current subset. You can unstar or unflag them as well. 
+
+You may wish to create a custom subset of your data through a series of separate faceting activities (rather than successively narrowing down with multiple facets applied). For example, you may wish to: 
+*   apply a facet
+*   star all the matching rows
+*   remove that facet 
+*   apply another, unrelated facet
+*   star all the new matching rows (which will not modify already-starred rows)
+*   remove that facet
+*   and then work with all of the cumulative starred rows. 
+
+You can use the dropdown menu on the “All” column and selecting “Facet by star” or “Facet by flag.” This will create “true” and “false” facets in the facet sidebar.
+
+You can also create a text facet on any column with the expression ```row.starred``` or ```row.flagged```.
+
+## Text filter
+
+Filters allow you to narrow down your data based on whether a given column includes a text string. 
+
+When you choose “Text filter” a box appears in the “Facet/Filter” sidebar that allows you to enter in text. Matching rows will narrow dynamically with every character you enter. You can set the search to be case-sensitive or not, and you can use this box to enter in a regular expression. This field supports [Java's regular expression language](http://download.oracle.com/javase/tutorial/essential/regex/).
+
+For example, you can employ a regular expression to view all properly-formatted emails:
```

Comment:
Is it worth giving a simple text filter example as well as a regular expression version?